pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -1716,8 +1716,52 @@ def clear(self):
         self.curve.clear()
         self.scatter.clear()
 
-    def appendData(self, *args, **kwargs):
-        pass
+
+    def appendData(self, new_x, new_y):
+        """
+        Append new data to the existing dataset.    
+        
+        Parameters
+        ----------
+        new_x : array-like
+            The new x-values to append.
+        new_y : array-like
+            The new y-values to append.
+
+        Warning
+        -------
+        This method may be significantly slower than calling `setData`, as it 
+        appends to the current data rather than replacing it. For high-performance
+        applications, consider using `setData` when possible.
+
+        This method will be undergoing further changes in future versions. 
+        Currently, the accepted arguments are only a subset of those for `setData`.
+        Ensure compatibility by using only the supported arguments.
+        
+        """
+        if self._dataset is None:
+            # If there is no existing dataset, set the new data as the dataset.
+            self.setData(new_x, new_y)
+            return
+        
+        # Convert new data to arrays
+        new_x = np.array(new_x)
+        new_y = np.array(new_y)
+
+        # Concatenate new data to existing dataset
+        xData = np.concatenate((self._dataset.x, new_x))
+        yData = np.concatenate((self._dataset.y, new_y))
+
+        # Update the internal dataset with the concatenated arrays
+        self._dataset.x = xData
+        self._dataset.y = yData
+
+        # Invalidate any display data cache
+        self._datasetMapped  = None
+        self._datasetDisplay = None  
+        self._adsLastValue   = 1
+        
+        self.updateItems(styleUpdate=False)  
```

Comment:
`self._datasetDisplay` is the cache for the processed data.

If you set that to `None`, doesn't the call to `updateItems()` go to `_getDisplayDataset()` and refreshes the cache by recalculating everything?

Could you tell me which part of the processing you are avoiding?

Reply:
We still need to clear the display cache (self._datasetDisplay) to accommodate new data points right?

Reply:
Yes.

Your first post said "without clearing and recalculating the plot", I am still trying to figure out where this manages to skip something that might be slow in a real-time application.

The automatic refresh of the cleared cache seems like it is still processing the entire dataset, after all. And I think the current design of the `PlotDataItem` internals do need that.

Reply:
In live data scenarios, recalculating everything with each update can indeed slow things down, especially if data transformations (like log scaling or other range calculations) are re-applied each time.